external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -203,7 +215,13 @@ class KafkaTestUtils(withBrokerProps: Map[String, Object] = Map.empty) extends L
 
   /** Add new partitions to a Kafka topic */
   def addPartitions(topic: String, partitions: Int): Unit = {
-    AdminUtils.addPartitions(zkUtils, topic, partitions)
+    val existingAssignment = zkClient.getReplicaAssignmentForTopics(
+      collection.immutable.Set(topic)).map {
+        case (topicPartition, replicas) => topicPartition.partition -> replicas
+    }
```

Comment:
We can get replica assignment information via AdminClient too. I think we should try to avoid the internal `ZkUtils` and `KafkaZkClient` as much as we can.

Reply:
+1